### ISSUE TITLE
How to specify `--platform` argument image pull option for multi platform images

### ISSUE THREAD

**Issue by vivanov83** [fabric8io/docker-maven-plugin#1532]
It's supported when doing build but I think while pulling images we're not passing this. We can fix this issue alongside this issue.

_Originally posted by @rohanKanojia in https://github.com/fabric8io/docker-maven-plugin/issues/1516#issuecomment-1011109432_

Does this part is fixed, or planed to be fixed? 
If it's fixed already, may I have an example how we're defining multiplatform image pull in xml

**Comment by vivanov83** [comment]
Yeah, I saw that, but this option is only for building/creating images, not for `pulling` them in FROM section defined in Dockerfile as an example : 
`FROM --platform=linux/arm64  debian:stretch-slim`
If my build platform is x86 based or x64 not ARM for example, the final image will be `x` based, and the only workaround I can see here is to define `parent` image to be the platform I / you want.Am I right ?

**Comment by rohanKanojia** [comment]
This option should work for pulling images. When you're pulling, you're creating image as well on your docker host.I think  you'd need to set platform to the platform you're current docker host is based (from where you're pulling it).

**Comment by rhuss** [comment]
@rohanKanojia @vivanov83 could you please fix the subject of this issue ?

**Comment by rohanKanojia** [comment]
@vivanov83 : polite ping, Did you get time to try my suggestion?

**Comment by vivanov83** [comment]
Let me explain differently : 
In order to build for example ARM image what you need : 
1 - st ARM based 'parent' image  (a.k.a FROM ) 
2 - nd provide parameters for docker to build arm based image(a.k.a docker build ... --platform ARM) 
If you don't specify --platform when you're doing FROM operation then docker daemon will pull the image based on the **host**  architecture. 
I one word/sentence . I would like to build ARM based images on x86-64 hosts, but if there is no way to pulling the proper `parent` image then this is not gonna work. 
Correct me if I'm wrong

**Comment by chonton** [comment]
Consider trying buildx support just added in #1548

**Comment by rohanKanojia** [comment]
buildx support should be available in 0.40.0 . Please try it out and provide us feedback!

**Comment by chonton** [comment]
@rohanKanojia please close

**Comment by AhHa45** [comment]
> buildx support should be available in 0.40.0 . Please try it out and provide us feedback!

I don't understand how this is related to the image pull...the buildx.platforms setting does not help.
(Doesn't even make sense since we're not building anything we just want to pull the right image)
I am on a apple macbook pro m1 (with an arm64 cpu) and get this warning:

[WARNING] DOCKER> The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested

how to pull the right image?!
```
<execution>
						<id>start-postgres</id>
						<phase>pre-integration-test</phase>
						<goals>
							<goal>start</goal>
						</goals>
						<configuration>
							<images>
								<image>
									<name>postgres:12.7</name>
									<alias>postgres</alias>
									<build>
										<buildx>
											<platforms>linux/arm64</platforms>
										</buildx>
									</build>
									<run>
										...
									</run>
								</image>
							</images>
						</configuration>

```

**Comment by chonton** [comment]
Docker will just pull the right image if you don't specify <buildx><platforms>

**Comment by AhHa45** [comment]
> Docker will just pull the right image if you don't specify

as you can see in the warning I posted, it does not.

**Comment by chonton** [comment]
SInce you have specified 
```
                    <buildx>
                        <platforms>linux/arm64</platforms>
                    </buildx>
```
only the `linux/arm64` platform will be pulled.  Since your macbook needs a `linux/arm64/v8` image, this warning appears.

If you remove the `platforms` section above, docker should be smart enough to pull the appropriate image.

**Comment by AhHa45** [comment]
confirmed, thx @chonton

**Comment by sta-szek** [comment]
hi, i faced the same issue, however the `buildx` did not help me for __RUN CONFIGURATION__

in my case, there is a single sentence in docs that pointed me to fix:

> platform | Specify an explicit platform to use when starting a docker container. May be set with property docker.platform. Defaults to native platform.

in given section: http://dmp.fabric8.io/#start-configuration

i think it could be a bit better documented, that those two options are different - one for build and the other for run. also, the docs contain exampel for build, but not for run.

also, as stated in docs:

> May be set with property docker.platform.

setting `<docker.plaform>` in maven properties, or `-Ddocker.platform` when running maven command does not work at all.

the only working option for me is to override parent plugin image configuration and specify like this:

```
<plugins>
    <plugin>
        <groupId>io.fabric8</groupId>
        <artifactId>docker-maven-plugin</artifactId>
        <configuration>
            <autoPull>always</autoPull>
            <images>
                <image>
                    <name>registry.dev.coresuite.com/postgresql-ms-test:latest</name>
                    <alias>postgres</alias>
                    <run>
                        <platform> linux/amd64</platform>
...
```

**Comment by liry** [comment]
FYI: I've created an issue to either make it work for runnig or fix the docs: #1750
